skeletons/classname.xml

```diff
@@ -1,105 +1,132 @@
-     </ooclass>
--->
- 
-<!-- If the class implements an interface, or two, use this (one oointerface element per interface)
-     <oointerface>
-      <interfacename>InterfaceName</interfacename>
-     </oointerface>
--->
-    </classsynopsisinfo>
-<!-- }}} -->
- 
-    <classsynopsisinfo role="comment">Properties</classsynopsisinfo>
-<!-- If the property is documented below (xml:id=classname.props) use this
+
+   <!-- {{{ Synopsis -->
+   <classsynopsis class="class">
+    <ooclass>
+     <classname>ClassName</classname>
+    </ooclass>
+
+    <!-- If the class extends another one, use this -->
+    <ooclass>
+     <modifier>extends</modifier>
+     <classname>ParentClass</classname>
+    </ooclass>
+
+    <oointerface>
+     <modifier>implements</modifier>
+     <interfacename>Interface1</interfacename>
+    </oointerface>
+
+    <oointerface>
+     <interfacename>Interface2</interfacename>
+    </oointerface>
+
+    <oointerface>
+     <interfacename>Interface3</interfacename>
+    </oointerface>
+
+    <classsynopsisinfo role="comment">&Constants;</classsynopsisinfo>
     <fieldsynopsis>
      <modifier>public</modifier>
+     <modifier>const</modifier>
+     <type>int</type>
+     <varname linkend="classname.constants.some-constant">ClassName::SOME_CONSTANT</varname>
+    </fieldsynopsis>
+
+    <classsynopsisinfo role="comment">&Properties;</classsynopsisinfo>
+    <fieldsynopsis>
+     <modifier>protected</modifier>
      <type>string</type>
-     <varname linkend="classname.props.name">name</varname>
+     <varname linkend="classname.props.someproperty">SomeProperty</varname>
+     <initializer>""</initializer>
     </fieldsynopsis>
--->
- 
-<!-- If properties are documented "as a method", use this
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:fieldsynopsis)" />
--->
 
-<!-- Edit the classname below -->
+    <!-- Edit the parentclass below -->
+    <classsynopsisinfo role="comment">&InheritedProperties;</classsynopsisinfo>
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.parentclass')/db:partintro/db:section/db:classsynopsis/db:fieldsynopsis[preceding-sibling::db:classsynopsisinfo[1][@role='comment' and text()='&Properties;']]))">
+     <xi:fallback/>
+    </xi:include>
+
+    <!-- Edit the classname below -->
+    <!-- If has a constructor -->
     <classsynopsisinfo role="comment">&Methods;</classsynopsisinfo>
-    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[not(@role='procedural')])">
-     <xi:fallback />
+    <xi:include xpointer="xmlns(db=http://docbook.org/ns/docbook) xpointer(id('class.classname')/db:refentry/db:refsect1[@role='description']/descendant::db:constructorsynopsis[@role='ClassName'])">
+     <xi:fallback/>
```

Comment:
Likely a good idea, but might be a small footgun for authors not updating `ClassName` here.